LGTM: This fixes the described issues. 

However, some other issues came up as discussed:
* `make bootstrapper` does not work as documented
* `bazel build //:scion-bootstrapper-deb` fails because of missing librpm:
```
$ bazel build //:scion-bootstrapper-deb
Starting local Bazel server and connecting to it...
WARNING: --enable_bzlmod is set, but no MODULE.bazel file was found at the workspace root. Bazel will create an empty MODULE.bazel file. Please consider migrating your external dependencies from WORKSPACE to MODULE.bazel. For more details, please refer to https://github.com/bazelbuild/bazel/issues/18958.
INFO: Analyzed target //:scion-bootstrapper-deb (96 packages loaded, 519 targets configured).
ERROR: /home/ztilmann/work/bootstrapper/BUILD.bazel:60:8: Writing: bazel-out/k8-fastbuild/bin/scion-bootstrapper-bin.tar failed: missing input file '//:bin/bootstrapper'
ERROR: /home/ztilmann/work/bootstrapper/BUILD.bazel:60:8: Writing: bazel-out/k8-fastbuild/bin/scion-bootstrapper-bin.tar failed: 1 input file(s) do not exist
Target //:scion-bootstrapper-deb failed to build
Use --verbose_failures to see the command lines of failed build steps.
ERROR: /home/ztilmann/work/bootstrapper/BUILD.bazel:60:8 Writing: bazel-out/k8-fastbuild/bin/scion-bootstrapper-bin.tar failed: 1 input file(s) do not exist
INFO: Elapsed time: 5.731s, Critical Path: 0.04s
INFO: 18 processes: 18 internal.
ERROR: Build did NOT complete successfully
```
* `make all` always executes
```
	bazel run //:gazelle -- update-repos -from_file=go.mod -to_macro=go_deps.bzl%go_deps -prune
	bazel clean --expunge
```
* `make all` fails with
```
$ make all
bazel run //:gazelle -- update-repos -from_file=go.mod -to_macro=go_deps.bzl%go_deps -prune
INFO: Analyzed target //:gazelle (1 packages loaded, 133 targets configured).
INFO: Found 1 target...
Target //:gazelle up-to-date:
  bazel-bin/gazelle-runner.bash
  bazel-bin/gazelle
INFO: Elapsed time: 0.289s, Critical Path: 0.02s
INFO: 1 process: 1 internal.
INFO: Build completed successfully, 1 total action
INFO: Running command line: bazel-bin/gazelle update-repos '-from_file=go.mod' '-to_macro=go_deps.bzl%go_deps' -prune
bazel clean --expunge
INFO: Starting clean (this may take a while). Consider using --async if the clean takes more than several minutes.
rm -f bin/*
./.bazel-build-env
STABLE_PKG_VERSION_SCION_BOOTSTRAPPER v0.0.0-v0.0.7-21-ged73f2d-dirty
bazel build //:bootstrapper
Starting local Bazel server and connecting to it...
INFO: Analyzed target //:bootstrapper (178 packages loaded, 13271 targets configured).
INFO: Found 1 target...
Target //:bootstrapper up-to-date:
  bazel-bin/bootstrapper
INFO: Elapsed time: 30.375s, Critical Path: 19.43s
INFO: 91 processes: 8 internal, 83 linux-sandbox.
INFO: Build completed successfully, 91 total actions
cp `bazel aquery  'outputs(".*bin/bootstrapper", //:bootstrapper)' --output=text 2>/dev/null | grep "Outputs" | sed -r 's/\s*Outputs: \[(.*)\]/\1/'` bin/
ln -sf ./bin/bootstrapper ./scion-bootstrapper
bazel test --config=unit --test_output=errors ...
ERROR: /home/ztilmann/work/bootstrapper/BUILD.bazel:133:8: in pkg_rpm rule //:scion-bootstrapper-rpm: 
Traceback (most recent call last):
	File "/home/ztilmann/.cache/bazel/_bazel_ztilmann/87be20ea61cc62702d6eca3dd1110526/external/rules_pkg/rpm.bzl", line 47, column 17, in _pkg_rpm_impl
		fail("The rpmbuild_toolchain is not properly configured: " +
Error in fail: The rpmbuild_toolchain is not properly configured: @@rules_pkg_rpmbuild//:rpmbuild_auto
ERROR: /home/ztilmann/work/bootstrapper/BUILD.bazel:133:8: Analysis of target '//:scion-bootstrapper-rpm' failed
Use --verbose_failures to see the command lines of failed build steps.
ERROR: Analysis of target '//:scion-bootstrapper-rpm' failed; build aborted
INFO: Elapsed time: 0.375s, Critical Path: 0.10s
INFO: 28 processes: 26 internal, 2 linux-sandbox.
ERROR: Build did NOT complete successfully
//config:go_default_test                                              NO STATUS
//fetcher:go_default_test                                             NO STATUS

Executed 0 out of 2 tests: 2 were skipped.
FAILED: 
    Fetching repository @@com_github_stretchr_testify; starting
make: *** [Makefile:61: test] Error 1
```

--- a/Makefile
+++ b/Makefile
@@ -30,7 +30,7 @@ install_deps:
 	  fi; \
 	else \
 	  exit 0; \
-	fi;
+	fi; \
 	apt-get install build-essential 2>/dev/null; \
 	test $$? -eq 0 || echo "Install build tools?\nsudo apt-get install build-essential" && sudo apt-get install build-essential
 

--- a/Makefile
+++ b/Makefile
@@ -1,36 +1,61 @@
 
-.PHONY: all bazel build clean realclean package package_deb package_rpm
+.PHONY: all bazel bootstrappper build go_build install_deps clean realclean package package_deb package_rpm
 
-build: bazel
-#go build -ldflags "-X github.com/netsec-ethz/bootstrapper/config.versionString="$(./.bazel-build-env | awk '{print $2}')
+build: install_deps bazel
+
+go_build:
+	@go build -o scion-bootstrapper -ldflags "-X github.com/netsec-ethz/bootstrapper/config.versionString="$$(git describe --tags --dirty --always)
+
+bootstrapper:
+	@if [ -n "$${OS}" ] ; then \
+	  UNAME_OS="$${OS}"; \
+	else \
+	  UNAME_OS=$$(uname --kernel-name); \
+	fi; \
+	if [ "$${UNAME_OS}" = "Linux" ] ; then \
+	  make -s build; \
+	elif [ "$${UNAME_OS}" = "Windows_NT" ] ; then \
+	  make.exe -s go_build; \
+	  cp scion-bootstrapper scion-bootstrapper.exe; \
+	else \
+	  make -s go_build; \
+	fi;
+
+install_deps:
+	apt-get install build-essential
 
 bazel: go_deps.bzl
 	rm -f bin/*
+	@if [ -z "$${CC}" ] ; then \
+	  if [ ! -x "$$(command -v "gcc")" ]; then \
+	    echo "Cannot find gcc or CC; set the CC environment variable or make sure gcc is on your PATH." && exit 1; \
+	  fi; \
+	fi;
 	./.bazel-build-env
 	bazel build //:bootstrapper
 	cp `bazel aquery  'outputs(".*bin/bootstrapper", //:bootstrapper)' --output=text 2>/dev/null | grep "Outputs" | sed -r 's/\s*Outputs: \[(.*)\]/\1/'` bin/
 	ln -sf ./bin/bootstrapper ./scion-bootstrapper
 
-all: build test package
+all: bootstrapper test package
 
 clean:
 	bazel clean
-	rm -f bin/*
+	rm -f ./bin/*
 
 realclean: clean
 	bazel clean --expunge
-	rm -r ~/.cache/bazel
-	rm -rf ./bazel-*
 	rm -f ./MODULE.bazel*
-	rm -f go_deps.bzl
+	rm -f ./go_deps.bzl
 
 package: package_deb
 
 package_deb: build
+	apt-get install python3
 	bazel build //:scion-bootstrapper-deb
 	cp bazel-bin/scion-bootstrapper_*_*.deb bin/
 
 package_rpm: build
+	apt-get install rpm
 	bazel build //:scion-bootstrapper-rpm
 	cp bazel-bin/scion-bootstrapper-*.*.rpm bin/
 
@@ -55,9 +80,16 @@ ifeq (,$(wildcard go_deps.bzl))
 	$(file > ./go_deps.bzl,$(go_deps_boilerplate))
 endif
 	bazel run //:gazelle -- update-repos -from_file=go.mod -to_macro=go_deps.bzl%go_deps -prune
-	rm -rf ./bazel-*
-	rm -r ~/.cache/bazel
+	bazel clean --expunge
 
 test: build
-	bazel test --config=unit --test_output=errors ...
+	@if [ ! -x "$$(command -v "scion-pki")" ]; then \
+	  echo "Cannot find scion-pki; make sure scion-pki is on your PATH otherwise the tests relying on it will fail."; \
+	fi;
+	bazel test --test_output=errors //hinting:go_default_test //fetcher:go_default_test //config:go_default_test
+	# Do not filter on unit tag for now, as doing so forces all loaded packages
+	# to satisfy their dependencies, which is not desirable for pkg_rpm
+	# which forces to have a full rpm toolchain loaded, even when the target
+	# is not executed. Explicitly list all test targets.
+	#bazel test --config=unit --test_output=errors ...
 

--- a/Makefile
+++ b/Makefile
@@ -22,7 +22,17 @@ bootstrapper:
 	fi;
 
 install_deps:
-	apt-get install build-essential
+	@if [ -z "$${CC}" ] ; then \
+	  if [ ! -x "$$(command -v "gcc")" ]; then \
+	    echo "Cannot find gcc or CC; set the CC environment variable or make sure gcc is on your PATH."; \
+	  else \
+	    exit 0; \
+	  fi; \
+	else \
+	  exit 0; \
+	fi;
+	apt-get install build-essential 2>/dev/null; \
+	test $$? -eq 0 || echo "Install build tools?\nsudo apt-get install build-essential" && sudo apt-get install build-essential
 
 bazel: go_deps.bzl
 	rm -f bin/*
@@ -50,12 +60,20 @@ realclean: clean
 package: package_deb
 
 package_deb: build
-	apt-get install python3
+	if [ ! -x "$$(command -v "python3")" ]; then \
+	  echo "Cannot find python3 on your PATH."; \
+	  apt-get install python3 &>/dev/null; \
+	  test $$? -eq 0 || echo "Install python3?\nsudo apt-get install python3" && sudo apt-get install python3
+	fi;
 	bazel build //:scion-bootstrapper-deb
 	cp bazel-bin/scion-bootstrapper_*_*.deb bin/
 
 package_rpm: build
-	apt-get install rpm
+	if [ ! -x "$$(command -v "rpmbuild")" ]; then \
+	  echo "Cannot find rpmbuild on your PATH."; \
+	  apt-get install rpm &>/dev/null; \
+	  test $$? -eq 0 || echo "Install rpm toolchain?\nsudo apt-get install rpm" && sudo apt-get install rpm
+	fi;
 	bazel build //:scion-bootstrapper-rpm
 	cp bazel-bin/scion-bootstrapper-*.*.rpm bin/
 

